src/js/utils/JobUtil.js

```diff
@@ -0,0 +1,90 @@
+import Job from '../structs/Job';
+
+const getMatchingProperties = function (job, item) {
```

Comment:
We should introduced this method in the `SchemaUtil` instead and is should be capable to simply  create modal from a schema and additionally augmenting it with data object like the structs in this example.  
As we don't have time at the moment, I'd like to focus on the actual feature and do this refactoring in the next dev cycle. What do you think?